ports/zephyr/prj_base.conf

```diff
@@ -14,9 +14,6 @@ CONFIG_NEWLIB_LIBC=y
 CONFIG_FLOAT=y
 CONFIG_MAIN_STACK_SIZE=4736
 
-# Drivers
-CONFIG_I2C=y
```

Comment:
My concern with doing this is that it forces a board-specific .conf file to enable a feature that exists on many many zephyr boards, and you'll end up adding a new .conf file for every single zephyr board you want to use in micropython. Instead, I think we can have:
1) a base `prj.conf` that enables i2c and the most common zephyr features. It does not have to be the least common denominator.
2) `boards/${BOARD}.conf` files that augment `prj.conf`. For example, enable the display on rt1050. I'm thinking about trying to get https://github.com/littlevgl/lv_binding_micropython working with zephyr.
3) `prj_${BOARD}.conf` for special cases like `qemu_cortex_m3` that want to bypass the base `prj.conf` entirely.

https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/app/boilerplate.cmake#L333-L352

Reply:
> My concern with doing this is that it forces a board-specific .conf file to enable a feature that exists on many many zephyr boards, and you'll end up adding a new .conf file for every single zephyr board

Yes, that's how it was conceived to be from the start, and how it is in MicroPython. Not hardware specific features are enabled in prj.conf, which gets merged with `prj_${BOARD}.conf`which contains hardware-specific setting for a particular board. And I2C is definitely a hardware-specific setting. Yes, that's simple and flat structure, and it may be improved. But that would be a separate change, which requires good thinking out, and I'd say, Zephyr config system itself would need to be improved. (That's the whole idea - MicroPython represents a non-trivial app, on which many Zephyr matters can be tested and validated).

> prj_${BOARD}.conf for special cases like qemu_cortex_m3 that want to bypass the base prj.conf entirely.

I don't think that there should be such cases. Zephyr is an OS which should work consistently on all boards. For extreme cases like not enough ROM, it should be possible to *disable* some feature in prj_${BOARD}.conf which was enabled in prj.conf.

> https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/app/boilerplate.cmake#L333-L352

I personally don't think that it's necessarily to blindly follow Zephyr things (because in many places they're subideal). Again, currently `prj_${BOARD}.conf` is overlay on top of prj.conf. I personally would classify act of putting them under boards/ as "hiding configs from users". (Obviously that won't scacle to hundreds of boards, but that's the point - there shouldn't be hundreds of configs, and I'd say Zephyr should grow to avoid that on its side).

All in all, this patch fixes regression as introduced by earlier changes, as caught by CI.